### ISSUE TITLE
terraform plugin incorrectly requires optional (conditional) parameters as mandatory

### ISSUE THREAD

**Issue by Ed42** [VladRassokhin/intellij-hcl#335]
# AWS Lambda function definition.

Terraform plugin incorrectly highlights as error if `handler` and `runtime` are omitted, even though the `package_type` is set to `Image`. Terraform will perform the initial apply with `handler` and `runtime` incorrectly present, but delta processing will fail with a Terraform message to the effect of: "do not supply handler and runtime with package_type set to Image"

### Installation details
 - [ ] IDE version
 IntelliJ IDEA 2021.1.1 (Ultimate Edition)
Build #IU-211.7142.45, built on April 30, 2021
Licensed to Edward Statham
Subscription is active until November 28, 2021.
Runtime version: 11.0.10+9-b1341.41 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Linux 5.11.0-7614-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
Registry: documentation.show.toolbar=true, debugger.watches.in.variables=false
Non-Bundled Plugins: Dilbert (1.9.0.168), com.intellij.ideolog (203.0.27.0), com.pixelperfect.gitlab-ci (1.0.6-211), org.intellij.plugins.hcl (0.7.10), name.kropp.intellij.makefile (211.6693.108), org.toml.lang (0.2.146.3826-211), GrepConsole (11.9.211.6086.0), io.codearte.props2yaml (1.0.2), org.avaje.ebean11.enhancer (12.8.2), cucumber-javascript (211.6693.65), Dart (211.7233), org.jetbrains.kotlin (211-1.5.0-release-759-IJ6693.72), Pythonid (211.7142.45), net.ashald.envfile (3.2.1), ru.adelf.idea.dotenv (2021.1), org.asciidoctor.intellij.asciidoc (0.32.43)
Kotlin: 211-1.5.0-release-759-IJ6693.72
Current Desktop: pop:GNOME

 - [ ] intellij-hcl plugin version `0.7.10`
 - [ ] Terraform version `0.14.2`

### Terraform Configuration Files
The `handler` and `runtime` properties should not be supplied with `package_type` set to `Image`. However if they are removed then the error highlight is triggered.
```hcl
resource "aws_lambda_function" "lambda" {
  function_name = "${local.resource_prefix}-${var.lambda_name}"
  image_uri    = "${var.image_repository_ecr}:${var.image_tag}"
  package_type = "Image"
  handler     = var.handler
  runtime     = var.runtime
  timeout     = var.timeout
  memory_size = var.memory_size
}
```

### Expected Behavior
plugin should correctly handle lambda definition without `handler` or `runtime` parameters in the case where the `package_type` is `Image`

### Actual Behavior
error highlight on lambda definition complaining about missing properties, when they are not in fact allowed with `package_type` set to `Image`

### Steps to Reproduce
Define a lambda function with `package_type` set to `Image`, and do *not* supply `handler` or `runtime` properties. Error highlight should appear. HCL snippet above should demonstrate this - place into a `.tf` file then remove or comment out the `runtime` and `handler` properties. The error highlight will (incorrectly) trigger.

**Comment by Ed42** [comment]
anyone?

**Comment by kunickiaj** [comment]
Running into this same one -- I've ended up disabling this inspection entirely for the meantime.

**Comment by VladRassokhin** [comment]
Fixed in 0.7.11
